transport-msg.tex

```diff
@@ -62,6 +62,27 @@ \subsection{Errors Definition}\label{sec::Virtio Transport Options / Virtio Over
 \hline
 \end{tabular}
 
+\subsection{Configuration Generation Count}\label{sec::Virtio Transport Options / Virtio Over Messages / Configuration Generation Count}
+
+To allow the driver to cache the configuration and prevent sending messages to
+access the configuration when it did not change, the Virtio Message Transport
+is using a \emph{Configuration generation count} value.
+
+The \emph{Configuration generation count} is a counter that is incremented by
+the device every time a value is modified in the configuration. It is
+communicated to the driver through an event (VIRTIO_MSG_EVENT_CONFIG) each time
+a configuration value is modified by the device itself and in the response to
+any modification of the configuration by the driver (through the
+VIRTIO_MSG_SET_CONFIG message).
+
+When the driver wants to modify a part of the configuration values, it has to
+provide the current value of the \emph{Configuration generation count} and the
+device will reject the request if the value provided is not the one it
+currently has.
+
+This system allows the driver to maintain a cache of the configuration space
+of a device and return values from it instead of asking the device when a
+configuration value is needed.
```

Comment:
I'm trying to understand how it work  from virtio driver perspective. 
let's say that the device update the configuration. Is there a specific mechanism  to notify the virtio driver from the virtio transport that the config has changed?

Reply:
When the device does an update in the configuration, it has to send an EVENT_CONFIG message to notify the driver that something changed in the config